packages/core/src/main/conversation/ConversationService.ts

```diff
@@ -916,9 +920,85 @@ export class ConversationService {
       payload = conversationData;
     }
 
+    if (typeof conversationData !== 'string' && conversationData.protocol === ConversationProtocol.MLS) {
+      return this.createMLSConversation(conversationData);
+    }
+
     return this.apiClient.api.conversation.postConversation(payload);
   }
 
+  private async createMLSConversation(conversationData: NewConversation): Promise<Conversation> {
+    /**
+     * @note For creating MLS conversations the users & qualified_users
+     * field must be empty as backend is not aware which users
+     * are in a MLS conversation because of the MLS architecture.
+     */
+    const newConversation = await this.apiClient.api.conversation.postConversationV2({
+      ...conversationData,
+      users: undefined,
+      qualified_users: undefined,
+    });
+    const {group_id: groupId} = newConversation;
+    const groupIdDecodedFromBase64 = Decoder.fromBase64(groupId!).asBytes;
+    const {qualified_users: qualifiedUsers = [], selfUserId} = conversationData;
+    if (!selfUserId) {
+      throw new Error('You need to pass self user qualified id in order to create an MLS conversation');
+    }
+
+    /**
+     * @note We need to fetch key packages for all the users
+     * we want to add to the new MLS conversations,
+     * includes self user too.
+     */
+    const keyPackages = await Promise.all([
+      this.apiClient.api.client.claimMLSKeyPackages(
+        selfUserId.id,
+        selfUserId.domain,
+        /**
+         * we should skip fetching key packages for current self client,
+         * it's already added by the backend on the group creation time
+         */
+        conversationData.creator_client,
+      ),
+      ...qualifiedUsers.map(qualifiedId =>
+        this.apiClient.api.client.claimMLSKeyPackages(qualifiedId.id, qualifiedId.domain),
+      ),
+    ]);
```

Comment:
*idea*: not sure if its cleaner to read but occurs to me we could just push the selfUser data into the qualifiedId list and then just spread one call inside the promise.all.

Reply:
yes but that way we're gonna need extra logic for conversationData.creator_client which we are passing as skip_own to not fetch key package for current self client

Reply:
ah yes i was thinking we could just pass it in once but it wouldn't work exactly that way. lets leave it as is